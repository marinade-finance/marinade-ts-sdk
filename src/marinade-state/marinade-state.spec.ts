import {  Marinade, MarinadeUtils, web3 } from '..'

describe('MarinadeState', () => {
  it('getStakeStates', async() => {
    const marinade = new Marinade()
    const state = await marinade.getMarinadeState()

    const accountInfos: {
      pubkey: web3.PublicKey
      account: web3.AccountInfo<Buffer>
    }[] = [{
      account: {
        data: Buffer.from('AgAAAIDVIgAAAAAANW0aj6LBKPbJQ/wTWTSoQgM4pWWci256Ye2TQjl8FVuAaQtLGkbornYeejJYw3kxRYbab8LmYo3X4irqaL1q7AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEClM1NCYixYJ/F511x003U5jAB5HWc+6vGBUWA4WISTCwRyggAAAACTAAAAAAAAAP//////////AAAAAAAA0D/YExAEAAAAAAAAAAA=', 'base64'),
        executable: false,
        lamports: 2190793099,
        owner: MarinadeUtils.STAKE_PROGRAM_ID,
        rentEpoch: 224,
      },
      pubkey: new web3.PublicKey('6yWLeYR8RsBHGbAvUGQhsi72JEhn2sZAjY2jxjQPT5sC'),
    }]

    marinade.provider.connection.getProgramAccounts = jest.fn().mockResolvedValueOnce(accountInfos)

    const stakeStates = await state.getStakeStates()

    expect(stakeStates).toMatchSnapshot()
  })
})
