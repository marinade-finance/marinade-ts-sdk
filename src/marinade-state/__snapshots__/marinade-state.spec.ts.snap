// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarinadeState getStakeStates 1`] = `
Array [
  StakeState {
    "Stake": StakeState {
      "meta": Meta {
        "authorized": Authorized {
          "staker": PublicKey {
            "_bn": "356d1a8fa2c128f6c943fc135934a8420338a5659c8b6e7a61ed9342397c155b",
          },
          "withdrawer": PublicKey {
            "_bn": "80690b4b1a46e8ae761e7a3258c379314586da6fc2e6628dd7e22aea68bd6aec",
          },
        },
        "lockup": Lockup {
          "custodian": PublicKey {
            "_bn": "00",
          },
          "epoch": "00",
          "unixTimestamp": "00",
        },
        "rentExemptReserve": "22d580",
      },
      "stake": Stake {
        "creditsObserved": "041013d8",
        "delegation": Delegation {
          "activationEpoch": "93",
          "deactivationEpoch": "ffffffffffffffff",
          "stake": "8272040b",
          "voterPubkey": PublicKey {
            "_bn": "40a5335342622c5827f179d75c74d375398c00791d673eeaf181516038588493",
          },
          "warmupCooldownRate": Object {
            "value": 0.25,
          },
        },
      },
    },
  },
]
`;
